Hatchery modal queue

rocketguy2 opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Please fill out this form, so that we have all the info needed to review your changes -->
<!-- Any text inside of the angle brackets will not show up in the final comment. Check the Preview tab to confirm -->

## Description
<!-- Describe your changes in detail -->
<!-- If you are adding new content, please let us know if it is based on some canon, and provide a reference to it -->

The Breeding Queue is now visible inside the Modal. In addition, an option is available to disable this in case the player wants to stick with how the modal used to look. The file breedingModal INLINETEST.html is an alternative take on what the queue could look like in the modal.

EDIT: The breeding queue is now below the pokemon list in the modal.


## Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- Please also link to any related issues or PRs here. -->
<!-- You can link an issue to be auto-closed when this is merged by writing 'Closes #1234' or 'Fixes #1234' -->

This change was made in order to make it so that the player does not have to leave the hatchery modal to make changes to the queue. I see no benefit to making the player have to exit and re-enter the modal if, for example, they accidentally put a pokemon in the queue that they did not want in there. I have added an alternative layout in case the other is preferred, and added a setting to disable the feature entirely if the player is so inclined.

## How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

I have filled and emptied the queue in the Hatchery successfully, and I have tested the setting to remove the queue from the modal, which worked.

## Screenshots (optional):
<!-- Drag a screenshot into this textbox to upload your image -->

Desktop

<img width="1593" height="863" alt="image" src="https://github.com/user-attachments/assets/a4e2348a-4d13-4b93-a3f4-ca0c309f7367" />
<img width="1574" height="839" alt="image" src="https://github.com/user-attachments/assets/43c6eeb3-5e8b-42ad-94f9-9f002e07bebb" />
<img width="1575" height="869" alt="image" src="https://github.com/user-attachments/assets/073efe7a-ab37-4274-89f1-511b874a4601" />

Mobile(ish)

<img width="969" height="779" alt="image" src="https://github.com/user-attachments/assets/0dd82277-c492-4e43-a1fe-1e0b1504825a" />
<img width="970" height="835" alt="image" src="https://github.com/user-attachments/assets/aa41ae9b-c0c1-4639-b864-c2f60ccde4ae" />
<img width="977" height="859" alt="image" src="https://github.com/user-attachments/assets/8f12225b-961c-407d-910d-d377cd55b8b7" />


## Types of changes
<!-- What types of changes does your code introduce? Delete any that don't apply, and/or add more you think would be useful -->

- UI improvement

rocketguy2 committed:
Added the Breeding Queue to the Breeding Modal

The Breeding Queue is now visible inside the Modal. In addition, an option is available to disable this in case the player wants to stick with how the modal used to look. The file breedingModal INLINETEST.html is an alternative take on what the queue could look like in the modal.

rocketguy2 committed:
Forgot a semicolon

whoops

RegisCoaxans commented:
It's hard to do change in the hatchery modal. There isn't much space left to play with.
We would like it better if the list was in a footer instead. It's better not to move the Pokémon list up and down. Also, this might fix an issue with the additional scrollbar on the Pokémon list. If it doesn't, then you'll have to (it's due to hardcoded height in [breeding.less](https://github.com/pokeclicker/pokeclicker/blob/5498b2802aac11cb4d321919c997e7a18668aa35/src/styles/breeding.less#L326)).
Lastly, could you make the setting *false* as a default, please ?

rocketguy2 committed:
Made the hatchery queue default to hidden in modal

rocketguy2 committed:
Moved the queue to the bottom of the modal

This moves the breeding queue from the top of the breeding modal to the bottom, and fixes a bug where on mobile the pokemon in the queue aren't visible.

rocketguy2 committed:
Merge branch 'develop' into HatcheryModalQueue

RegisCoaxans committed:
Merge branch 'develop' into HatcheryModalQueue

RegisCoaxans committed:
Cleanup
